src/ccnl-core/src/ccnl-relay.c

```diff
@@ -903,9 +903,7 @@ ccnl_fib_rem_entry(struct ccnl_relay_s *relay, struct ccnl_prefix_s *pfx,
 
     if (fwd) {
         if (fwd->face) {
-            if (&fwd->face->peer) {
```

Comment:
I think the check is unnecessary as well. The member ``peer`` is of type ``sockunion`` and hence no pointer. What basically happens is that that the member ``peer`` is returned via ``fwd->face->peer`` and than the address of that very member is taken, which is rather pointless since it is always set and valid, see also
```C
 32 struct ccnl_face_s {
 33     struct ccnl_face_s *next, *prev;
 34     int faceid;
 35     int ifndx;
 36     sockunion peer;
 37     int flags;
 38     int last_used; // updated when we receive a packet
 39     struct ccnl_buf_s *outq, *outqend; // queue of packets to send
 40     struct ccnl_frag_s *frag;  // which special datagram armoring
 41     struct ccnl_sched_s *sched;
 42 #ifdef CCNL_RIOT
 43     evtimer_msg_event_t evtmsg_timeout;
 44 #endif
 45 };
```
If you really want to check if the member is set (and hence the debug statement makes sense) you probably would need to initialise the struct with ``-1`` or ``0`` and check in the if statement if the value of the member is still ``-1`` or ``0``.

Reply:
`peer` seems to be a nested struct in `ccnl_face_s`. So it can actually never be `NULL` if `face` points to something